lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemInputFile.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.URIStatus;
+import alluxio.exception.AlluxioException;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.runtime.NotFoundRuntimeException;
+import alluxio.grpc.OpenFilePOptions;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoInput;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Optional;
+
+import static io.trino.filesystem.alluxio.AlluxioUtils.convertToAlluxioURI;
+import static java.util.Objects.requireNonNull;
+
+public class AlluxioFileSystemInputFile
+        implements TrinoInputFile
+{
+    private final Location location;
+
+    private final FileSystem fileSystem;
+
+    private final String mountRoot;
+    private Optional<Instant> lastModified;
+
+    private Long length;
+
+    private URIStatus status;
+
+    public AlluxioFileSystemInputFile(Location location, Long length, FileSystem fileSystem, String mountRoot, Optional<Instant> lastModified)
+    {
+        this.location = requireNonNull(location, "location is null");
+        this.fileSystem = requireNonNull(fileSystem, "fileSystem is null");
+        this.mountRoot = requireNonNull(mountRoot, "mountRoot is null");
+        this.length = length;
+        this.lastModified = requireNonNull(lastModified, "lastModified is null");
+    }
+
+    @Override
+    public TrinoInput newInput()
+            throws IOException
+    {
+        try {
+            return new AlluxioFileSystemInput(openFile(), this);
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error newInput() file: %s".formatted(location), e);
+        }
+    }
+
+    @Override
+    public TrinoInputStream newStream()
+            throws IOException
+    {
+        try {
+            return new AlluxioTrinoInputStream(location, openFile(), getStatus());
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error newStream() file: %s".formatted(location), e);
+        }
+    }
+
+    private FileInStream openFile()
+            throws IOException, AlluxioException
+    {
+        if (!exists()) {
+            throw new FileNotFoundException("File does not exist: " + location);
+        }
+        return fileSystem.openFile(getStatus(), OpenFilePOptions.getDefaultInstance());
+    }
+
+    private URIStatus getStatus(boolean lazy)
```

Comment:
Make this similar to `io.trino.filesystem.hdfs.HdfsInputFile#loadFileStatus` by returning void and populating both length and lastModified in this method

Reply:
resolved